examples/azure/terraform-azure-automationaccount-example/main.tf

```diff
@@ -0,0 +1,197 @@
+# ---------------------------------------------------------------------------------------------------------------------
+# DEPLOY AN AZURE AUTOMATION ACCOUNT, AN AZURE VM ALONG WITH AN EXAMPLE DESIRED STATE CONFIGURATION (DSC)
+# This is an example of how to deploy an Automation Account along with a basic DSC to support applying and
+# enforcing a configuration on a basic virtual machine.
+# ---------------------------------------------------------------------------------------------------------------------
+# See test/azure/terraform_azure_automationaccount_example_test.go for how to write automated tests for this code.
+# ---------------------------------------------------------------------------------------------------------------------
+
+provider "azurerm" {
+  version = "~>2.20"
+  features {}
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# PIN TERRAFORM VERSION TO >= 0.12
+# The examples have been upgraded to 0.12 syntax
+# ---------------------------------------------------------------------------------------------------------------------
+
+terraform {
+  required_version = ">= 0.12"
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# DEPLOY A RESOURCE GROUP
+# See test/terraform_azure_nsg_example_test.go for how to write automated tests for this code.
+# ---------------------------------------------------------------------------------------------------------------------
+
+resource "azurerm_resource_group" "automation_account_dsc_rg" {
+    name     = "${var.resource_group_name}-${var.postfix}"
+    location = var.location
+}
+
+resource "random_password" "virtual_machine" {
+  length           = 16
+  override_special = "-_%@"
+  min_upper        = "1"
+  min_lower        = "1"
+  min_numeric      = "1"
+  min_special      = "1"
+}
+
+resource "azurerm_automation_account" "automation_account" {
```

Comment:
consider removing  automation_account_name varibales and set the  name = "terratest-AutomationAccount-${var.postfix}" . like how the rest of the azure examples are implemented.

Reply:
It's a best practice in Terraform to not hard code properties / names for resources and use variables in main.tf.  There are several Azure examples that use variables as well:
terraform-azure-vm-example
terraform-azure-nsg-example
terraform-azure.sqldb-example

I'd prefer to use variables and not hard-code all property values but we if we want to hard-code all property values it should be consistent for all examples.  Should I update the others as well?